pose_to_tf/src/main.cpp

```diff
@@ -14,32 +14,48 @@
  * the License.
  */
 
-#include <ros/ros.h>
+#include <rclcpp/rclcpp.hpp>
 #include <tf2_ros/transform_broadcaster.h>
-#include <pose_to_tf/PoseToTF2.h>
-#include <pose_to_tf/PoseToTF2Config.h>
-#include <carma_utils/CARMANodeHandle.h>
+#include <pose_to_tf/PoseToTF2.hpp>
+#include <pose_to_tf/PoseToTF2Config.hpp>
+#include <carma_ros2_utils/carma_lifecycle_node.hpp>
 
 int main(int argc, char** argv)
 {
-  ros::init(argc, argv, "pose_to_tf");
-
-  ros::CARMANodeHandle nh;
-  ros::CARMANodeHandle pnh("~");
-
-  tf2_ros::TransformBroadcaster tf_broadcaster;
 
+  rclcpp::init(argc, argv);
+  auto node = std::make_shared<rclcpp::Node>("pose_to_tf");
+ 
   pose_to_tf::PoseToTF2Config config;
-  pnh.param<std::string>("child_frame", config.child_frame, config.child_frame);
-  pnh.param<std::string>("default_parent_frame", config.default_parent_frame, config.default_parent_frame);
+  // Create default config
+  //config_ = Config();
+  //Declare parameters and defaults
+  config.child_frame = declare_parameter<std::string>("message_type", config.child_frame);
+  config.default_parent_frame = declare_parameter<std::string>("target_frame", config.default_parent_frame);
```

Comment:
You should not be loading your configuration in the main function. It should be loaded in handel_on_configure like our other ros2 nodes. This will allow for proper lifecycle behavior. Additionally, all our ros2 nodes are launched as components which would never run the main function. 

Rather than having PoseToTF take in the node, you can make it extend the carma base node just like the other ros2 nodes in the system.